Mobile/ShopLens/Droid/Source/ShoppingListActivity.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using Android.App;
+using Android.OS;
+using Android.Widget;
+
+namespace ShopLens.Droid
+{
+    [Activity(Label = "ShoppingListActivity")]
+    public class ShoppingListActivity : Activity
+    {
+        EditText addItemEditText;
+        Button addItemButton;
+        ListView listView;
+
+        List<string> items;
+
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+            SetContentView(Resource.Layout.ShoppingList);
+
+            listView = FindViewById<ListView>(Resource.Id.ShopListListView);
+            addItemButton = FindViewById<Button>(Resource.Id.ShopListAddItemButton);
+            addItemEditText = FindViewById<EditText>(Resource.Id.ShopListAddItemEditText);
+
+            items = new List<string> { "Coconut", "Banana", "Rice", "Beer" };
+            ArrayAdapter<string> listAdapter = 
+                new ArrayAdapter<String>(this, Android.Resource.Layout.SimpleListItemChecked, items);
+            listView.Adapter = listAdapter;
+            listView.ChoiceMode = ChoiceMode.Multiple;
+
+            addItemButton.Click += (sender, e) =>
+            {
+                if (!String.IsNullOrWhiteSpace(addItemEditText.Text))
+                {
+                    listAdapter.Add(addItemEditText.Text);
+                }
+                listAdapter.NotifyDataSetChanged();
```

Comment:
In here I imagine "NotifyDataSetChanged" updates the `listView` elements?

Reply:
It notifies the adapter that it's data set has been modified and it should refresh the listview that the adapter fills with data.